crates/rpc-types-engine/src/envelope.rs

```diff
@@ -203,6 +229,32 @@ impl OpNetworkPayloadEnvelope {
         Ok(Self { payload, signature, payload_hash: hash, parent_beacon_block_root: None })
     }
 
+    /// Encodes a payload envelope as a snappy-compressed byte array.
+    #[cfg(feature = "std")]
+    pub fn encode_v2(&self) -> Result<Vec<u8>, PayloadEnvelopeEncodeError> {
+        let execution_payload_v2 = match &self.payload {
+            OpExecutionPayload::V2(execution_payload_v2) => execution_payload_v2,
+            _ => return Err(PayloadEnvelopeEncodeError::WrongVersion),
+        };
+
+        use ssz::Encode;
+        let mut data = Vec::new();
+        let mut sig_bytes = self.signature.as_bytes();
+        // Normalize parity bit
+        if self.signature.v() {
+            sig_bytes[64] = 0x01;
+        } else {
+            sig_bytes[64] = 0x00;
+        }
```

Comment:
Small nit - let's use `Parity::y_parity_byte` here & in the other parity bit normalizations: https://docs.rs/alloy-primitives/latest/alloy_primitives/enum.Parity.html#method.y_parity_byte

```suggestion
        sig_bytes[64] = self.signature.v().y_parity_byte();
```

Reply:
Oh nice!